src/main/java/com/faforever/client/replay/ReplayDetailController.java

```diff
@@ -221,20 +221,27 @@ public void setReplay(Replay replay) {
             () -> ratingLabel.setText("-"));
 
     if (replay.getReplayFile() == null) {
-      replayService.getSize(replay.getId())
-          .thenAccept(replaySize -> JavaFxUtil.runLater(() -> {
-            if (replaySize > -1) {
-              String humanReadableSize = Bytes.formatSize(replaySize, i18n.getUserSpecificLocale());
-              downloadMoreInfoButton.setText(i18n.get("game.downloadMoreInfo", humanReadableSize));
-              watchButton.setText(i18n.get("game.watchButtonFormat", humanReadableSize));
-              downloadMoreInfoButton.setVisible(true);
-            } else {
-              downloadMoreInfoButton.setText(i18n.get("game.replayFileMissing"));
-              downloadMoreInfoButton.setDisable(true);
-              watchButton.setText(i18n.get("game.replayFileMissing"));
-              watchButton.setDisable(true);
-            }
-          }));
+      if (replay.getReplayAvailable()) {
+        replayService.getSize(replay.getId())
+            .thenAccept(replaySize -> JavaFxUtil.runLater(() -> {
+              if (replaySize > -1) {
+                String humanReadableSize = Bytes.formatSize(replaySize, i18n.getUserSpecificLocale());
+                downloadMoreInfoButton.setText(i18n.get("game.downloadMoreInfo", humanReadableSize));
+                watchButton.setText(i18n.get("game.watchButtonFormat", humanReadableSize));
+                downloadMoreInfoButton.setVisible(true);
+              } else {
+                downloadMoreInfoButton.setText(i18n.get("game.replayFileMissing"));
+                downloadMoreInfoButton.setDisable(true);
+                watchButton.setText(i18n.get("game.replayFileMissing"));
+                watchButton.setDisable(true);
+              }
+            }));
+      } else {
+        downloadMoreInfoButton.setText(i18n.get("game.replayNotAvailable"));
+        downloadMoreInfoButton.setDisable(true);
+        watchButton.setText(i18n.get("game.replayNotAvailable"));
+        watchButton.setDisable(true);
+      }
```

Comment:
The replayFileMissing case will no longer occur due to replaySize. It's not 100% guaranteed but 99,99%.
Also maybe it would be nice to distinguish betweeen "not available" and "not yet available". E.g. if the game finished over an hour ago I would not expect it to show up at all, but if it is younger than (or if no scoreTime set) it is probably still in processing.

Reply:
Alright so the scoreTime is only set once the file is written? Or I might just do the hour.
And if the 99.99% is good then I can just remove the other logic just wasnt sure

Reply:
No, probably not. The scoreTime is written when the lobby server has receive all game ended messages. That is not necessarily the same moment the replayserver wrote the file. I think this requires the game to be closed, but I'm not sure.